build: Setup custom linting for applications, design-system etc

sharmayatendra opened the pull request:
This PR introduces custom linting setup across the entire project to enforce consistent code quality and formatting.

### root packages:
--- 

- Scoped packages to improve organization and prevent naming conflicts.
- Updated prettier configuration files to align with scoped packages.
- Added linting command for convenient code analysis and adherence to linting rules.


### next-app:
--- 

- Configured linting for the "next-app" package using scoped packages.
- Added linting command script to the package file for easy access.
- Formatted files such as _app, index, and _document according to the linting configuration.
- Scoped the package in the tsconfig for better isolation and encapsulation.

### next-next-app:
---

- Applied similar linting changes as done in the "next-app" package.

### design-system:
---

- Added a type attribute to the button component for improved type safety.
- Added linting command to facilitate code analysis and adherence to linting rules.
- Scoped packages to ensure modularity and minimize potential conflicts.
- Included necessary configurations (eslintignore, prettierignore, prettierrc) for linting and code formatting.

### eslint-config-custom:
--- 

- Configured necessary files (eslintignore, prettierignore, prettierrc) for consistent linting and formatting.
- Added eslint configuration to the index file to enforce linting rules and maintain code quality.

By implementing this custom linting setup, the project benefits from improved code quality, consistent formatting, and adherence to best practices. These changes enhance maintainability, readability, and collaboration among developers, leading to a more robust and reliable codebase.

turbobot-temp committed:
feat(create-turbo): apply official-starter transform

sharmayatendra committed:
build: Create initial turbo configuration

This commit introduces the initial configuration for the turborepo project. The configuration is set up using the npx create-turbo command, which creates the necessary files and dependencies to get started.

The following changes have been made:

eslint-config-custom: This package includes custom configurations for eslint, tailored for next, prettier, react, and turbo.

tsconfig: Compiler options have been added to the base, nextJS, and react-library files to ensure proper TypeScript compilation.

ui: Dependencies for React, TypeScript, and eslint have been included. Additionally, the Button and Header components have been added.

For further guidance, you can refer to the tutorial link which provides a video tutorial on setting up the turbo configuration.

sharmayatendra committed:
feat: Initiate Next projects

This commit focuses on creating two Next.js projects using the create-next-app command. The projects are named next-app and next-next-app, and they serve as starting points for building Next.js applications.

The following changes have been made:

In next-app:

Dependencies: Node.js, Next.js, React, and TypeScript dependencies have been added to support the project.

Next.js Configuration: The necessary configurations for Next.js have been set up.

PostCSS Config File: A PostCSS configuration file has been included.

Components: The initial set of components, including app, document, and others, have been added.

Tailwind Config File: A configuration file for Tailwind CSS has been included.

tsconfig: Compiler options have been added to the tsconfig file.

readme: Readme file is also added

In next-next-app:

Same files and folders as next-app have been added.

The application's name is specified as next-next-app, distinguishing it from next-app.

sharmayatendra committed:
build: Add shared TypeScript configs

This commit focuses on introducing shared TypeScript configuration settings that can be used across the project. By centralizing these configurations, we ensure consistency and facilitate easier maintenance and updates.

The changes made are as follows:

In next-app:

Extends the TypeScript configuration from the Next.js config file.

Modifies the compilerOptions to align with the project's requirements.
In tsconfig:

Adds additional compiler options to the base-config section.

Adds compilerOptions specific to the server-config section.

In server:

Adds express dependencies, which are required for building server-side functionality.

Adds nodemon, node, typescript, and @types/express as devDependencies, providing the necessary tools for development.

Implements a basic express application.

Defines an interface for the express app.

Additionally, some changes have been made to the prettier configuration file.

These shared TypeScript configurations enhance code consistency and improve the development experience throughout the project. It ensures that all parts of the project follow the same TypeScript settings and enables easier collaboration between team members.

sharmayatendra committed:
build: Add shared eslint configs

This commit focuses on adding shared ESLint configurations to the project, enabling consistent and standardized linting across the codebase. Linting helps improve code quality, identifies potential issues, and enforces coding best practices.

The following changes have been made:

In eslint-config-custom:

Added various dependencies such as airbnb, airbnb-base, and plugins like eslint-plugin.

Configured linting rules specifically tailored for Next.js projects.

Configured linting rules for the server code.
In next-app:

Implemented project-specific linting setup.

Modified the code according to the linting rules.
This includes using arrow-functions components and ensuring that React is properly in scope.

In next-next-app:

Applied the same linting configuration as next-app.

By adding shared ESLint configurations, we ensure consistent code quality, adhere to coding standards, and promote maintainable code across the project. Linting rules help catch common mistakes and improve code readability, facilitating collaboration within the development team.

sharmayatendra committed:
build: Add shared tailwind ui

This commit focuses on incorporating shared Tailwind UI components into the project. Tailwind CSS is a utility-first CSS framework that helps streamline the development of user interfaces.

The following changes have been made:

In the ui section:

Added devDependencies for tailwindcss, postcss, and autoprefixer. These are necessary for using Tailwind CSS in the project.

In next-app:

Added next-transpile-module as a devDependency. This module enables transpilation of specific modules within Next.js projects.

Configured next-app to utilize next-transpile-module.

Utilized the Button component from the shared Tailwind UI in the project.

By incorporating shared Tailwind UI components, we enhance the project's user interface development. Tailwind CSS simplifies styling and offers a wide range of pre-designed components, such as buttons, that can be easily integrated into the project.

sharmayatendra committed:
refactor: function declaration, package imports, scoped packages etc.

This commit focuses on refactoring various files by replacing arrow functions with function declarations and introducing scoped packages. Additionally, improvements have been made to package imports, transpiling modules, and code consistency using editor ruler settings.

Server:
Scoped the package to enhance modularity and prevent naming conflicts.

Next-app:
Replaced arrow functions with function declarations for improved readability and maintainability.

Scoped packages to ensure encapsulation and prevent naming clashes.

Introduced transpile modules to optimize module bundling.

Utilized editor ruler settings to maintain consistent code formatting and adherence to style guidelines.

UI (Design system):
Renamed "ui" to "design-system" to accurately reflect its purpose and intent.

Transitioned from arrow functions to function declarations for code consistency and standardization.

Scoped packages to promote encapsulation and minimize potential conflicts.

By making these refactorings, the codebase becomes more organized, easier to read, and aligned with established best practices. These changes improve code maintainability, enhance modularity, and contribute to a more robust and scalable project structure.

sharmayatendra committed:
build: Setup custom linting for applications, design-system etc

This commit introduces custom linting setup across the entire project to enforce consistent code quality and formatting.

Root packages:

Scoped packages to improve organization and prevent naming conflicts.

Updated prettier configuration files to align with scoped packages.

Added linting command for convenient code analysis and adherence to linting rules.

Next-app:

Configured linting for the "next-app" package using scoped packages.

Added linting command script to the package file for easy access.

Formatted files such as _app, index, and _document according to the linting configuration.

Scoped the package in the tsconfig for better isolation and encapsulation.

Next-next-app:

Applied similar linting changes as done in the "next-app" package.

Design-system:

Added a type attribute to the button component for improved type safety.

Added linting command to facilitate code analysis and adherence to linting rules.

Scoped packages to ensure modularity and minimize potential conflicts.

Included necessary configurations (eslintignore, prettierignore, prettierrc) for linting and code formatting.

Eslint-config-custom:

Configured necessary files (eslintignore, prettierignore, prettierrc) for consistent linting and formatting.

Added eslint configuration to the index file to enforce linting rules and maintain code quality.

Sowed commented on apps/next-app/src/pages/index.tsx:
```diff
@@ -0,0 +1,175 @@
+      </div>
+
+      <div
+        className="before:bg-gradient-radial after:bg-gradient-conic relative flex
```
move this class to a new line and index it. Something like...
```html
<div
   ...
        className="
          before:bg-gradient-radial 
          ...
        "
    ...
```

Sowed replied:
Note: For such long class name with lots of longer class names, and selectors. You should consider adding [`twin.macro` ](https://github.com/ben-rogerson/twin.macro)and use it with [`styled-components`](https://styled-components.com/).
- Then make a wrapper with container styled-component.
- See the `apps-suite` mono repo for inspiration.

Sowed replied:
Make this a point in the next tasks please.

sharmayatendra replied:
moved to the new line.

Sowed commented on apps/next-app/src/pages/index.tsx:
```diff
@@ -0,0 +1,175 @@
+
+      <div className="mb-32 grid text-center lg:mb-0 lg:grid-cols-4 lg:text-left">
+        <a
+          href="https://nextjs.org/docs?utm_source=create-next-app&utm_medium=
```
Sort the props. Always. Please.
Check the apps-suites config for this setup. You can look at `dealer-mobile/dev` for the latest config. This will have the linter call you our or do it for you on save.

Sowed commented on apps/next-app/src/pages/index.tsx:
```diff
@@ -0,0 +1,175 @@
+          href="https://vercel.com/templates?framework=next.js&utm_source=
+          create-next-app&utm_medium=default-template-tw&utm_campaign=
+          create-next-app"
+          className="group rounded-lg border border-transparent px-5 py-4
```
Again fix all these indents. Refer to the previous comments.

sharmayatendra replied:
fixed

Sowed commented on apps/next-next-app/src/pages/index.tsx:
```diff
@@ -0,0 +1,171 @@
+export default function Home() {
+  return (
+    <main
+      className={`flex min-h-screen flex-col items-center justify-between p-24 
```
Fix indent on class names.

sharmayatendra replied:
fixed

Sowed commented on apps/next-next-app/src/pages/index.tsx:
```diff
@@ -0,0 +1,171 @@
+
+const inter = Inter({ subsets: ['latin'] });
+
+export default function Home() {
```
Please add doc comments to all the Components and functions in the code. All of them

sharmayatendra replied:
added

Sowed commented on apps/next-next-app/src/pages/index.tsx:
```diff
@@ -0,0 +1,171 @@
+      </div>
+
+      <div
+        className="before:bg-gradient-radial after:bg-gradient-conic relative flex
```
Refer to my previous comments on indents. Do it across all items in this PR

Sowed commented on packages/design-system/Header.tsx:
```diff
@@ -0,0 +1,5 @@
+import * as React from 'react';
```
React import is not required for React 18. Remove this and any other cases across all files. Add doc comments too.

sharmayatendra replied:
removed

Sowed reviewed (commented):
@sharmayatendra, I have left you some comments. Can you please look at them? 
Mostly about doc comments and linting indents

divyansh0908nl commented on apps/next-app/src/pages/_app.tsx:
```diff
@@ -0,0 +1,7 @@
+import type { AppProps } from 'next/app';
+
+export default function App({ Component, pageProps }: AppProps) {
+  // eslint-disable-next-line react/jsx-props-no-spreading
```
why this comment?

sharmayatendra replied:
it was because of previous linting setup removed now

divyansh0908nl commented on packages/eslint-config-custom/index.js:
```diff
@@ -1,11 +1,152 @@
+      },
     },
   },
 };
```
add an extra line at the end

sharmayatendra replied:
done

divyansh0908nl commented on packages/design-system/tsconfig.json:
```diff
@@ -1,5 +1,8 @@
+    "jsx": "react-jsx"
+  },
   "exclude": ["dist", "build", "node_modules"]
 }
```
add an extra line at the end

divyansh0908nl commented on packages/design-system/package.json:
```diff
@@ -5,19 +5,17 @@
   "scripts": {
-    "lint": "eslint \"**/*.ts*\"",
+    "format": "prettier --write .",
+    "lint": "eslint \"**/*.ts*\" && yarn format",
```
also include js file in the lint command

divyansh0908nl commented on packages/design-system/index.tsx:
```diff
@@ -1,5 +1,5 @@
-import * as React from "react";
+import * as React from 'react';
 
```
no need to import react

sharmayatendra replied:
removed

divyansh0908nl commented on packages/design-system/Button.tsx:
```diff
@@ -1,9 +1,11 @@
   return (
-    <button className="bg-blue-500 hover:bg-blue-700 h-16 w-16">Boop</button>
-  )
+    <button type="button" onClick={() => alert('boop')}>
```
remove this inline callback function, instead accept this as a prop

sharmayatendra replied:
done

divyansh0908nl commented on packages/design-system/Button.tsx:
```diff
@@ -1,9 +1,11 @@
+'use client';
 
-import * as React from 'react'
+import * as React from 'react';
```
remove react import as it is not needed

divyansh0908nl commented on package.json:
```diff
@@ -1,22 +1,23 @@
     "build": "turbo run build",
-    "dev": "turbo run dev",
-    "lint": "turbo run lint",
-    "format": "prettier --write \"**/*.{ts,tsx,md}\""
```
also include this command for .js file extensions

sharmayatendra replied:
done

divyansh0908nl commented on apps/server/.eslintrc.js:
```diff
@@ -1,7 +1,7 @@
     project: './tsconfig.json',
   },
-}
+};
```
leave an extra line at the end

divyansh0908nl commented on apps/next-next-app/next.config.js:
```diff
@@ -2,6 +2,6 @@
+};
 
-module.exports = nextConfig
+module.exports = nextConfig;
```
extra line

divyansh0908nl commented on apps/next-app/package.json:
```diff
@@ -21,5 +21,8 @@
     "typescript": "5.1.3"
+  },
+  "devDependencies": {
+    "next-transpile-modules": "^10.0.0"
```
look if this module is needed?

sharmayatendra replied:
removed

divyansh0908nl commented on apps/next-app/next.config.js:
```diff
@@ -1,6 +1,5 @@
-const nextConfig = {
-  reactStrictMode: true,
-}
+const withTM = require('next-transpile-modules')(['ui'])
```
look if this is needed

sharmayatendra replied:
removed

divyansh0908nl commented on packages/ui/tailwind.config.js:
```diff
@@ -0,0 +1,16 @@
+    },
+  },
+  plugins: [],
+}
```
an extra line at the end

divyansh0908nl commented on apps/next-next-app/src/pages/api/hello.ts:
```diff
@@ -1,13 +1,13 @@
 type Data = {
-  name: string
-}
+  name: string;
```
add doc

sharmayatendra replied:
added

sharmayatendra committed:
refactor: Add doc comments, removed unnecessary package, fixed indentation

- Removed `next-transpile-module` dependency.
- Add doc comments across for components and functions.
- Fixed indentation across the whole project.
- Add the `.js` extension as well for lint script.
